modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/poll/PollingSourceWrapper.java

```diff
@@ -351,31 +358,34 @@ private boolean passesWatermark(DefaultPollItem pollItem) {
               Serializable previousItemWatermark = recentlyProcessedIds.retrieve(itemId);
               if (compareWatermarks(itemWatermark, previousItemWatermark, watermarkComparator) <= 0) {
                 accept = false;
+              } else {
+                recentlyProcessedIds.remove(itemId);
```

Comment:
You are never comparing the item watermark to the updated watermark.

Let's say that currentWatermark is 5, and updatedWatermark is 10.

Maybe recentlyProcessed has the items with id A and watermark 5, and the same item A but with id 7 comes for processing, you are changing the updated watermark to 7. In the next poll, an item of watermark 8 would be accepted.